### PR TITLE
fix(language-core): prevent circular reference of templateRef

### PR DESCRIPTION
**Pull request by zhiyuanzmj**

### CODE REVIEW

File: test-workspace/tsc/passedFixtures/vue3.5/templateRef/template-ref.vue
Comment (KazariEX) [inline test-workspace/tsc/passedFixtures/vue3.5/templateRef/template-ref.vue:1]: I think it's best to keep consistency in code style.

File: packages/language-core/lib/codegen/script/template.ts [hunk 1]
```diff
@@ -257,3 +261,15 @@ export function getTemplateUsageVars(options: ScriptCodegenOptions, ctx: ScriptC
 
 	return usageVars;
 }
+
+function getRefsType(options: ScriptCodegenOptions, scriptSetupRanges: ScriptSetupRanges) {
```
Comment (KazariEX) [inline packages/language-core/lib/codegen/script/template.ts:265]: Can we use the generator here?

Comment (zhiyuanzmj) [reply packages/language-core/lib/codegen/script/template.ts:265]: I tried. But `const exps = []` is a string[] type, Can you try it?

Comment (KazariEX) [reply packages/language-core/lib/codegen/script/template.ts:265]: I found that `generateTemplateCtx` also needs to be converted into a pure generator-based form. In this case, its readability doesn't seem to differ much from using array operations :)

**Review by KazariEX — APPROVED** [review]

**Comment by johnsoncodehk** [comment]
Thanks!

**Pull request merged**